GVFS/GVFS.Common/NuGetUpgrade/OrgNuGetUpgrader.cs

```diff
@@ -0,0 +1,244 @@
+               fileSystem,
+               config,
+               nuGetFeed,
+               credentialStore)
+        {
+            this.httpClient = httpClient;
+            this.platform = platform;
+        }
+
+        public override bool SupportsAnonymousVersionQuery { get => true; }
+
+        private OrgNuGetUpgraderConfig Config { get => this.nuGetUpgraderConfig as OrgNuGetUpgraderConfig;  }
+        private string OrgInfoServerUrl { get => this.Config.OrgInfoServer; }
+        private string Ring { get => this.Config.UpgradeRing; }
+
+        public static bool TryCreate(
+            ITracer tracer,
+            PhysicalFileSystem fileSystem,
+            LocalGVFSConfig gvfsConfig,
+            HttpClient httpClient,
+            ICredentialStore credentialStore,
+            bool dryRun,
+            bool noVerify,
+            out OrgNuGetUpgrader upgrader,
+            out string error)
+        {
+            OrgNuGetUpgraderConfig upgraderConfig = new OrgNuGetUpgraderConfig(tracer, gvfsConfig);
+            upgrader = null;
+
+            if (!upgraderConfig.TryLoad(out error))
+            {
+                upgrader = null;
+                return false;
+            }
+
+            if (!upgraderConfig.IsConfigured(out error))
+            {
+                return false;
+            }
+
+            if (!upgraderConfig.IsReady(out error))
+            {
+                return false;
+            }
+
+            string platform = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? "macOS" : "windows";
+
+            upgrader = new OrgNuGetUpgrader(
+                ProcessHelper.GetCurrentProcessVersion(),
+                tracer,
+                fileSystem,
+                httpClient,
+                dryRun,
+                noVerify,
+                upgraderConfig,
+                ProductUpgraderInfo.GetAssetDownloadsPath(),
+                platform,
+                credentialStore);
+
+            return true;
+        }
+
+        public override bool TryQueryNewestVersion(out Version newVersion, out string message)
+        {
+            newVersion = null;
+
+            if (!AzDevOpsOrgFromNuGetFeed.TryParseOrg(this.Config.FeedUrl, out string orgName))
+            {
+                message = "OrgNuGetUpgrader is not able to parse org name from NuGet Package Feed URL";
+                return false;
+            }
+
+            OrgInfoApiClient infoServer = new OrgInfoApiClient(this.httpClient, this.OrgInfoServerUrl);
+
+            try
+            {
+                this.highestVersionAvailable = infoServer.QueryNewestVersion(orgName, this.platform, this.Ring);
+            }
+            catch (HttpRequestException exception)
+            {
+                message = string.Format("Network error: could not connect to server ({0}). {1}", this.OrgInfoServerUrl, exception.Message);
+                this.TraceException(exception, nameof(this.TryQueryNewestVersion), $"Error connecting to server.");

```

Comment:
Not a big deal, but here and in some of the catches below we are using the `$` string formatter with no variables in the string.  Not sure if you want to leave that in.